content/blog/2019/06/2019-06-14-performance-testing-jenkins.adoc

```diff
@@ -0,0 +1,129 @@
+---
+layout: post
+title: "Performance Testing Framework for Jenkins Plugins"
+tags:
+- jenkins
+- jmh
+- plugins
+- benchmark
+- performance
+- gsoc
+author: abhyudayasharma
+---
+
+= Performance Testing Framework for Jenkins Plugins
+
+My work for the first phase of my Google Summer of Code project to improve the performance of
+the Role Strategy Plugin involved creating a framework for running benchmarks in Jenkins Plugins.
+We chose link:http://openjdk.java.net/jeps/230[Java Microbenchmark Harness] for running the benchamrks.
+
+The framework works by starting a temporary Jenkins instance for each fork of the JMH benchmark, just
+like `JenkinsRule` from Jenkins Test Harness.
+
+== Creating Benchmarks
+
+Benchmarks are run directly from your JUnit Tests which allows you to fail builds on the fly and easily run
+benchmarks from your IDE, just like unit tests. You can easily configure your benchmarks by either using your
+Java methods, or by using Jenkins Configuration-as-Code plugin and passing the path to your YAML file.
```

Comment:
An example would be great (maybe just as a link)

Reply:
I've added a section for Configuration as Code, is that enough or add should I add a link to a benchmark using CasC in Role Strategy.